core/src/main/java/org/apache/iceberg/CachingCatalog.java

```diff
@@ -29,24 +34,99 @@
 import org.apache.iceberg.catalog.Namespace;
 import org.apache.iceberg.catalog.TableIdentifier;
 import org.apache.iceberg.exceptions.AlreadyExistsException;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
-
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class that wraps an Iceberg Catalog to cache tables.
+ * <p>
+ * When {@code expirationIntervalMillis} is positive, the cache will expire tables after
```

Comment:
we can link to the explanation using `{@link CatalogProperties#TABLE_CACHE_EXPIRATION_INTERVAL_MS}` instead of duplicating the explanation in 2 places.

Reply:
Done.